lib/connection_manager.go

```diff
@@ -220,18 +234,18 @@ func (cmgr *ConnectionManager) subFromGroupKey(na *wire.NetAddress) {
 
 func (cmgr *ConnectionManager) getRandomAddr() *wire.NetAddress {
 	for tries := 0; tries < 100; tries++ {
-		// Lock the address map since multiple threads will be trying to read
-		// and modify it at the same time.
-		cmgr.mtxConnectedOutboundAddrs.RLock()
 		addr := cmgr.AddrMgr.GetAddress()
-		cmgr.mtxConnectedOutboundAddrs.RUnlock()
-
 		if addr == nil {
 			glog.V(2).Infof("ConnectionManager.getRandomAddr: addr from GetAddressWithExclusions was nil")
 			break
 		}
 
-		if cmgr.connectedOutboundAddrs[addrmgr.NetAddressKey(addr.NetAddress())] {
+		// Lock the address map since multiple threads will be trying to read
+		// and modify it at the same time.
+		cmgr.mtxConnectedOutboundAddrs.RLock()
+		ok := cmgr.connectedOutboundAddrs[addrmgr.NetAddressKey(addr.NetAddress())]
+		cmgr.mtxConnectedOutboundAddrs.RUnlock()
+		if ok {
```

Comment:
What has changed about the threading model that we need to make things thread-safe in this way?

Reply:
This was a bug. The mutex is meant for `connectedOutboundAddrs`